### PR TITLE
[WIP] TargetedLocationDataProcessor implementation

### PR DESCRIPTION
**Pull request by kamcio96**
You should check it ;)

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+public class TargetedLocationDataProcessor implements DataProcessor<TargetedLocationData, ImmutableTargetedLocationData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof Entity;
+    }
+
+    @Override
+    public Optional<TargetedLocationData> from(DataHolder dataHolder) {
+        if(dataHolder instanceof EntityLiving) {
```
Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java]: TagetedLocation should also consider `EyeofEnder`s, `Compasses`, etc.

Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java]: `Compass`es all point to the same location (as far as each client is concerned; each client can be given a different compass target). It's not per-`ItemStack`, it's per-`Player`.

Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java]: Right, for Players, you might want to set the fake "spawn location" then for the compass to point.

Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java]: @gabizou But what if you have multiple compasses in your hotbar? You can see both, but they would have to point in the same direction, regardless of the position each one is set to.

Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java]: @JBYoshi If you have multiple compasses, they use the "respawn" location for the player, however, we can use the `TargetedLocationData` as I've been suggesting for a while now.

File: src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    @Override
+    public ImmutableTargetedLocationData createImmutable() {
+        return null; // new ImmutableSpongeTargetedLocationData() ?
+    }
+
+    @Override
+    public Optional<TargetedLocationData> createFrom(DataHolder dataHolder) {
+        return from(dataHolder); //something more?
```
Comment (Aaron1011) [inline src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java]: You always want to return a DataMapiulator here if the data holder is applicable. The difference between this method and `from` is that while `from` can return absent when passed a supported data holder (if no pathfinding path is set, for example), this method shouldn't.

File: src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java (hunk 1 above)
Comment (JBYoshi) [inline src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java]: @gabizou I think I'm just misunderstanding what you said. It seemed like you said that `TargetedLocationData` would apply to `ItemStack`s containing compasses.

Comment (gabizou) [inline src/main/java/org/spongepowered/common/data/processor/data/TargetedLocationDataProcessor.java]: @JBYoshi `TargetedLocationData` does have some javadocs as to the "expected" use cases:
```
An {@link DataManipulator} handling the supposed targetted
{@link Location}. Usually for the case of {@link EnderPearl}s, the
targeted {@link Location} is where the {@link EnderPearl} will
move towards until it's expiration time. In the case of
{@link ItemStack}s of type {@link ItemTypes#COMPASS}, the targeted
{@link Location} is where the compass will point towards.
```

**Comment by gabizou** [comment]
You also need to register in the `SpongeGameRegistry` and in the `KeyRegistry`.

**Comment by gabizou** [comment]
Please update implementation.

**Comment by kamcio96** [comment]
I want create new PR for this data. Where minecraft store player's compass target?

**Comment by JBYoshi** [comment]
Minecraft doesn't store the compass target on the server. It just sends a packet to the client when the spawn position changes.

**Comment by kamcio96** [comment]
Thanks, I will use mixin ;)

**Pull request closed without merging**